### PR TITLE
feat(miniflare): add CF-Worker header to outgoing fetch requests

### LINKED ISSUE #4367: cf-worker header in local development
In production, Cloudflare Workers attaches the following header to all outgoing requests:

```
cf-worker: <slug>.workers.dev
```

This is extremely useful for identifying the worker making the request, but is also used by various services to block workers. That's totally understandable on their part, but when developing in local mode, **this header is not attached**. 

This can result in some pretty gnarly experiences when building an application entirely locally, everything working as expected, and then deploying to production workers and being hit with 403s or other errors from external services now that `cf-worker` is attached as a header.

One quick example is some Steam endpoints:

```bash
$ curl -I "https://steamcommunity.com/openid/login?openid.mode=check_authentication"
HTTP/1.1 200 OK
```

vs in workers prod:

```bash
$ curl -I "https://steamcommunity.com/openid/login?openid.mode=check_authentication" --header "cf-worker: foo"
HTTP/1.1 403 Forbidden
```

Is this something `workerd` should emulate when running locally? Or perhaps miniflare/wrangler to better represent a real world environment?

### PR DESCRIPTION
**Pull request by penalosa**
Fixes #4367.

Adds the `CF-Worker` header to outgoing fetch requests in Miniflare's local development environment to match production behavior. In production, Cloudflare adds this header to subrequests made by Workers, with the value set to the zone/domain.

The implementation reuses the existing `strip-cf-connecting-ip` wrapper worker that already intercepts outbound requests. A new optional `zone` option is exposed to allow specifying the header value. Wrangler now passes the zone value (extracted from the configured routes/host) to Miniflare automatically.

**Header value behavior:**
- If `zone` option is provided: uses that value (e.g., `example.com`)
- If `zone` is not provided but `name` is: uses `${name}.example.com` (e.g., `my-worker.example.com`)
- If neither is provided: uses `worker.example.com`

**Updates since last revision:**
- Wrangler now passes the zone value to Miniflare via `ConfigBundle.zone`, extracted from `dev.origin.hostname` (which comes from routes configuration)

**Human review checklist:**
- [ ] Verify that `event.config.dev?.origin?.hostname` is the correct source for the zone value in Wrangler
- [ ] Consider if there are edge cases where the origin hostname might differ from the actual zone

---

- Tests
  - [x] Tests included/updated
  - [ ] Automated tests not possible - manual testing has been completed as follows:
  - [ ] Additional testing not necessary because:
- Public documentation
  - [ ] Cloudflare docs PR(s):
  - [x] Documentation not necessary because: This aligns local behavior with existing production behavior that is already documented.

Link to Devin run: https://app.devin.ai/sessions/3c5448d105c44a5fbbc723f94bbf35b8
Devin PR requested by @penalosa

---

<a href="https://app.devin.ai/review/cloudflare/workers-sdk/pull/12186">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by penalosa** [comment]
The Cf-worker header should be set to the zone. Miniflare should expose an (optional) option to pass in a zone, which Wrangler will use. Additionally, Miniflare should default cf-worker to ${worker-name}.example.com if no zone is passed in

**Comment by penalosa** [comment]
Devin, please make this change

**Comment by penalosa** [comment]
Devin, it doesn't look like Wrangler passes the `zone` value in this PR

File: packages/miniflare/src/workers/core/strip-cf-connecting-ip.worker.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
+interface Env {
+	CF_WORKER_ZONE: string;
+}
+
 export default {
-	fetch(request) {
+	fetch(request, env: Env) {
```
Comment (NuroDev) [inline packages/miniflare/src/workers/core/strip-cf-connecting-ip.worker.ts:6]: Nit pick: Not needed if we're using `satisfies ExportedHandler<Env>`
```suggestion
	fetch(request, env) {
```

**Review by NuroDev — APPROVED** [review]

File: packages/miniflare/src/plugins/core/index.ts
[the remark this answers is not in the document]
Comment (penalosa) [reply packages/miniflare/src/plugins/core/index.ts:1]: This is fine. `stripCfConnectingIp` really means "this miniflare instance is a proxy and shouldn't do weird things to requests" which applies to both `stripCfConnectingIp` and adding a CF Worker header
